test/types/type-provider.test-d.ts

```diff
@@ -194,3 +194,224 @@ expectAssignable(server.withTypeProvider<TypeBoxProvider>().get(
     expectType<boolean>(req.body.z)
   }
 ))
+
+// -------------------------------------------------------------------
+// TypeBox Reply Type
+// -------------------------------------------------------------------
+
+expectAssignable(server.withTypeProvider<TypeBoxProvider>().get(
+  '/',
+  {
+    schema: {
+      response: {
+        200: Type.String(),
+        400: Type.Number(),
+        500: Type.Object({
+          error: Type.String()
+        })
+      }
+    }
+  },
+  async (_, res) => {
+    res.send('hello')
+    res.send(42)
```

Comment:
I know that it's is near the impossible due to the complexity, but, there is a way to define the response type after a `res.code()` is set? By default, the response should be the `schema.response['200']`, however in this case for instance:

`res.send(42)` is matching a 400 status code that is not sent in fact.